libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -521,6 +529,63 @@ bool AC_Fence::pre_arm_check(char *failure_msg, const uint8_t failure_msg_len) c
     return true;
 }
 
+/*
+  get our altitude in the FENCE_ALT_TYPE frame
+  return false if not available
+ */
+bool AC_Fence::get_alt_in_frame(float &alt)
+{
+    const auto &ahrs = AP::ahrs();
+
+    ahrs.get_relative_position_D_home(alt);
+
+    // translate to +ve up
+    alt = -alt;
+
+    /*
+      convert to desired altitude frame
+     */
+    const auto alt_frame = (Location::AltFrame)_alt_type.get();
+    switch (alt_frame) {
```

Comment:
I'm not understanding why we need this.

Can't we just construct a location object and call `get_alt(alt_frame)`?

Reply:
it needs to work with no location, eg. copter indoors

Reply:
If we're using Fence then Copters, Rovers, etc need to know their Location.  We require that the EKF origin has been set and once this is done, the vehicle knows its Location.

Reply:
Ah, sorry, in the case where only the altitude fence is specified the vehicle can arm without a Location.  We should consider enhancing the Copter pre-arm check so that if the altitude fence is enabled and the alt-type is set to Terrain then we also require a Location